src/ast-resolver.ts

```diff
@@ -27,7 +34,10 @@ export class ASTResolver {
 
   tryResolveNode<T extends NodeType>(nodeType: T, id: number): NodeTypeMap[T] | undefined {
     try {
-      return this.resolveNode(nodeType, id);
+      const node = this.resolveNode('*', id);
+      if (isNodeType(nodeType, node)) {
+        return node;
+      }
```

Comment:
To be honest this change ended up not being necessary, so we could revert it, but there is an explanation behind it.

What happened was that when I first introduced the namespace logic the transpiler slowed down significantly. To transpile OZ Contracts took perhaps 1 minute. So in debugging I traced it down to `resolveContract`/`tryResolveNode` and experimented with approaches to improve this.

The issue was that `deref: ASTDereferencer`, used by `tryResolveNode`, had an optimized path that worked only when the requested node type was correct. But `tryResolveNode` was being used through `resolveContract` to _check_ if a node id corresponded to a ContractDefinition. If the id was instead something like a StructDefinition, `deref` would fall back to brute force enumeration of all nodes in the source code to try to find one with this id (!).

The reason I introduced the `'*'` feature in ASTDerefencer was to avoid falling back to the slow path, instead quickly returning whatever node was found even if it was not the wanted node type.

This helped somewhat but I ended up finding a [much better optimization](https://github.com/OpenZeppelin/solidity-ast/blob/0d83a1e249a291c6c40e186cb49426d7785a4a5a/src/ast-dereferencer.ts#L31-L34) that causes this to not really matter.